service/src/main/java/uk/gov/hmcts/reform/fpl/service/docmosis/GatekeepingOrderGenerationService.java

```diff
@@ -0,0 +1,88 @@
+    DocmosisTemplateDataGeneration<DocmosisStandardDirectionOrder> {
+    private final CaseDataExtractionService dataService;
+
+    public DocmosisStandardDirectionOrder getTemplateData(CaseData caseData) {
+        GatekeepingOrderEventData eventData = caseData.getGatekeepingOrderEventData();
+        HearingBooking firstHearing = caseData.getFirstHearingOfType(HearingType.CASE_MANAGEMENT)
+            .orElse(null);
+
+        SaveOrSendGatekeepingOrder saveOrSendGatekeepingOrder = eventData.getSaveOrSendGatekeepingOrder();
+
+        JudgeAndLegalAdvisor judgeAndLegalAdvisor = getSelectedJudge(
+            eventData.getGatekeepingOrderIssuingJudge(), caseData.getAllocatedJudge()
+        );
+
+        DocmosisStandardDirectionOrder.DocmosisStandardDirectionOrderBuilder<?, ?> orderBuilder =
+            DocmosisStandardDirectionOrder.builder()
+                .judgeAndLegalAdvisor(dataService.getJudgeAndLegalAdvisor((judgeAndLegalAdvisor)))
+                .courtName(dataService.getCourtName(caseData.getCaseLocalAuthority()))
+                .familyManCaseNumber(caseData.getFamilyManCaseNumber())
+                .ccdCaseNumber(formatCCDCaseNumber(caseData.getId()))
+                .complianceDeadline(caseData.getComplianceDeadline())
+                .children(dataService.getChildrenDetails(caseData.getAllChildren()))
+                .respondents(dataService.getRespondentsNameAndRelationship(caseData.getAllRespondents()))
+                .respondentsProvided(isNotEmpty(caseData.getAllRespondents()))
+                .applicantName(dataService.getApplicantName(caseData.getAllApplicants()))
+                .directions(buildDirections(caseData.getGatekeepingOrderEventData()))
+                .hearingBooking(dataService.getHearingBookingData(firstHearing))
+                .crest(getCrestData());
+
+        if (SEALED.equals(saveOrSendGatekeepingOrder.getOrderStatus())) {
+            orderBuilder.courtseal(getCourtSealData());
+            orderBuilder.dateOfIssue(formatLocalDateToString(saveOrSendGatekeepingOrder.getDateOfIssue(), DATE));
+        } else {
+            orderBuilder.draftbackground(getDraftWaterMarkData());
+            orderBuilder.dateOfIssue("<date of issue TBA>");
+        }
+        return orderBuilder.build();
+    }
+
+    private List<DocmosisDirection> buildDirections(GatekeepingOrderEventData eventData) {
+        //add future directions here
+        return buildCustomDirections(eventData.getSdoDirectionCustom());
+    }
+
+    private List<DocmosisDirection> buildCustomDirections(List<Element<CustomDirection>> elements) {
+        List<CustomDirection> customDirections = unwrapElements(elements);
+        List<DocmosisDirection> formattedDirections = new ArrayList<>();
+        for (CustomDirection direction : customDirections) {
+            formattedDirections.add(DocmosisDirection.builder()
+                .assignee(direction.getAssignee())
+                .title(direction.getTitle())
```

Comment:
in this ticket title does not include direction compliance date (this will be fixed in follow up ticket)

Reply:
fixed in dev/standard-directions